src/streamlink_cli/argparser.py

```diff
@@ -1084,6 +1084,15 @@ def build_parser():
         Example: "https://hostname:port/"
         """
     )
+    http.add_argument(
+        "--websocket-proxy",
```

Comment:
this doesn't seems like an universal solution,
`--http-proxy` or `--https-proxy` might be easier.

Reply:
Depends - on the one I think it’s good for it to be more configurable. In case you want the WebSocket proxied but not the http connection, I didn’t try it but it might actually be better in the ustream case. What about `wss` does that use the `https-proxy` option?  

To be honest I’m quite undecided about it - I commented on the PR as soon as I made it suggesting the possibility of using `http-proxy`. Part of me thinks `http-proxy‘ should cover everything, http, https, and WebSocket. Life would be easier then, it would make it easier for regular users too. Then make it so you can override or unset the other options, of course this would be a breaking change...


Maybe it is better to just use `http-proxy` for this ... :|

Reply:
I think the distinction between `http-proxy` and `https-proxy` (and therefore also `websocket-proxy`) is unnecessary. I highly doubt that using different proxies for these parameters is a proper use case. Using different proxies is only useful if you know which requests Streamlink makes, so you can target certain URLs with different proxies. The distinction between the protocols alone doesn't make sense.

Reply:
Which is better? 

1. `http-proxy` for `ws://` and `https-proxy` for `wss://`.
2. Same as 1., but the default for `https-proxy` is the same as `http-proxy`.
3. Use `http-proxy` for both `ws://` and `wss://`.

I personally prefer 2. as it would kill two stones with one bird, there have been a few issues where people have needed to set `https-proxy` as well as `http-proxy`. As @bastimeyer mentioned, you have to know which connections streamlink will make - and sometimes we don’t even know which it will make, there could be redirects etc. 2. gives you a sensible default, but the ability to override it if needed, perhaps if your http proxy only supports https over https and http over http.

Reply:
The second point sounds like the best solution, as it's a non-breaking change. It should be done in a second PR, though.

Reply:
A separate PR indeed. There is a breaking change though, before `https-proxy` would be unset. I don’t think it’s a problem though :)